pkg/translator/aggregator.go

```diff
@@ -285,14 +381,14 @@ func (rg *ingressRuleGroup) toHTTPRoute() (gatewayv1beta1.HTTPRoute, []error) {
 		}
 
 		var numWeightedBackends, totalWeightSet int32
-		for _, path := range paths {
+		for _, path := range paths.Paths {
 			backendRef, err := toBackendRef(path.path.Backend)
 			if err != nil {
 				errors = append(errors, err)
 				continue
 			}
-			if path.extra != nil && path.extra.canary != nil && path.extra.canary.weight != 0 {
-				weight := int32(path.extra.canary.weight)
+			if path.annotations != nil && path.annotations.canaryGroup != nil && path.annotations.canaryGroup.weight != 0 {
```

Comment:
Why the rename from `extra` to `annotations`? I'm not sure that all custom extensions we'd ever want to support will be limited to annotations.